### PR TITLE
Feature: enable multi-k calculation for CUDA version of module_gint

### LINKED ISSUE #4853: LCAO GPU support
### Details

Instructions on setting `ks_solver` for LCAO basis calculation using GPU should be added.
https://abacus.deepmodeling.com/en/latest/advanced/acceleration/cuda.html

### Have you read the online manual http://abacus.deepmodeling.com/en/latest/

- [X] Yes, I have read the online manual.

### Task list for Issue attackers (only for developers)

- [ ] Identify the specific section of the documentation with the issue.
- [ ] Investigate the issue and determine the root cause.
- [ ] Research best practices and potential solutions for the identified issue.
- [ ] Update the documentation to address the issue, following the suggested improvement.
- [ ] Ensure the updated documentation adheres to the project's documentation standards.
- [ ] Test the updated documentation to ensure it is clear and accurate.
- [ ] Review and incorporate any relevant feedback from users or developers.
- [ ] Publish the updated documentation and notify the issue reporter.

### PR DESCRIPTION
**Pull request by dzzz2001**
This PR enables multi-k calculation for CUDA version of module_gint. fix #4853 

### Unit Tests and/or Case Tests for my changes
- two case tests `930_NO_BI2SE2CU2O2_k_GPU` and `934_NO_Si2_dzp_neq_ns2_k_GPU` are added for multi-k calculation.

### CODE REVIEW

**Review by mohanchen — APPROVED** [review]

**Comment by dzzz2001** [comment]
fix #4853

File: docs/advanced/acceleration/cuda.md [hunk 1]
```diff
@@ -32,16 +32,13 @@ Check the [Advanced Installation Options](https://abacus-rtd.readthedocs.io/en/l
 ## Run with the GPU support by editing the INPUT script:
 
 In `INPUT` file we need to set the value keyword [device](../input_files/input-main.md#device) to be `gpu`.
```
Comment (caic99) [inline docs/advanced/acceleration/cuda.md]:
```suggestion
- Set `device`: In `INPUT` file we need to set the input parameter [device](../input_files/input-main.md#device) to `gpu`. If this parameter is not set, ABACUS will try to determine if there are available GPUs.
```

File: docs/advanced/acceleration/cuda.md [hunk 2]
```diff
@@ -32,16 +32,13 @@ Check the [Advanced Installation Options](https://abacus-rtd.readthedocs.io/en/l
...
+- **ks_solver**: The setting of keyword [ks_solver](../input_files/input-main.md#ks_solver) differs between PW basis and LCAO basis. For the PW basis, ks_solver is set by default to `cg`, but you can also manually set it to `bpcg`, `dav`, or `dav_subspace`. For the LCAO basis set, ks_solver defaults to `scalapack` (if ELPA is not compiled and mpi is enabled). Note that regardless of whether device is set to `cpu` or `gpu`, ks_solver is always defaulted to `scalapack`. If you wish to use `cusolver` for solving, you can manually set ks_solver to `cusolver`.
```
Comment (caic99) [inline docs/advanced/acceleration/cuda.md]:
```suggestion
- Set `ks_solver`: For the PW basis, CG, BPCG and Davidson methods are supported on GPU; set the input parameter [ks_solver](../input_files/input-main.md#ks_solver) to `cg`, `bpcg` or `dav`. For the LCAO basis, `cusolver` is supported on GPU.
```

File: docs/advanced/input_files/input-main.md [hunk 3]
```diff
@@ -629,7 +629,7 @@ If only one value is set (such as `kspacing 0.5`), then kspacing values of a/b/c
   - cpu: for CPUs via Intel, AMD, or Other supported CPU devices
   - gpu: for GPUs via CUDA or ROCm.
 
-  Known limitations: If using the pw basis, the ks_solver must be cg/bpcg/dav to support `gpu` acceleration. If using the lcao basis, `gamma_only` must be set to `1`, as multi-k calculation is currently not supported for `gpu`. lcao_in_pw also does not support `gpu`.
+  Known limitations: If using the pw basis, the ks_solver must be cg/bpcg/dav to support `gpu` acceleration. lcao_in_pw currently does not support `gpu`.
```
Comment (caic99) [inline docs/advanced/input_files/input-main.md]:
```suggestion
  Known limitations: `ks_solver` must also be set to the algorithms supported. lcao_in_pw currently does not support `gpu`.
```

File: docs/advanced/acceleration/cuda.md [hunk 4]
```diff
@@ -4,7 +4,7 @@ In ABACUS, we provide the option to use GPU devices to accelerate performance. T
 
 - **Full gpu implementations**: During the SCF progress, `Psi`, `Hamilt`, `Hsolver`, `DiagCG`, and `DiagoDavid` classes are stored or calculated by the GPU devices.
 
-- **Electronic state data**: (e.g. electronic density) are moved from the GPU `to the CPU(s) every scf step.
+- **Electronic state data**: (e.g. electronic density) are moved from the GPU to the CPU(s) every scf step.
 
 - **Acclerated by the NVIDIA libraries**: `cuBLAS` for common linear algebra calculations, `cuSolver` for eigen values/vectors, and `cuFFT` for the conversions between the real and recip spaces.
```
Comment (caic99) [inline docs/advanced/acceleration/cuda.md]: typo here: accelerated

**Pull request merged**